Part A. Visitors to your website are asked to answer a single survey question before they get access to the content on the page. Among all of the users, there are two categories: Random Clicker (RC), and Truthful Clicker (TC). There are two possible answers to the survey: yes and no. Random clickers would click either one with equal probability. You are also giving the information that the expected fraction of random clickers is 0.3. After a trial period, you get the following survey results: 65% said Yes and 35% said No. What fraction of people who are truthful clickers answered yes? Hint: use the rule of total probability.

Part B. Imagine a medical test for a disease with the following two attributes:

The sensitivity is about 0.993. That is, if someone has the disease, there is a probability of 0.993 that they will test positive.
The specificity is about 0.9999. This means that if someone doesn't have the disease, there is probability of 0.9999 that they will test negative.
In the general population, incidence of the disease is reasonably rare: about 0.0025% of all people have it (or 0.000025 as a decimal probability).
Suppose someone tests positive. What is the probability that they have the disease?

### Answers

#### Part A

Random and Truthful
Random 50-50 30% of the data
65% said yes, 35% said no

Let the total be x
Random =0.3x
Yes_confirmed =0.5*0.3x
No confirmed = 0.5*0.3x
Total yes = 0.65x
Remaining Yes = 0.65x-0.15x == 0.5x
Total No =0.35x
Remaining No =0.2x
Total truthful = 0.7x
Fraction = 0.5*100/0.7

In [1]:
0.5*0.3

0.15

In [2]:
0.36-0.15

0.21

In [3]:
print(round(0.5*100/0.7,3),'%')

71.429 %


The fraction of people who are truthful clickers answered yes are 71.43%

##### Part B

TP= 0.993
TN = 0.9999

A: Someone Tests Positive
B: Someone has the disease

P(B) = 0.000025 (prob of having the disease).
P(A|B) = 0.993 (prob of testing positive if they have the disease).
P(~A|~B) = 0.9999 (prob of testing negative if they dont have the disease).
To Find - Prob of having the disease if they are positive. Therefore P(B|A).

Bayes' theorem:
P(B|A) = P(A|B)*P(B)/P(A)


Given P(A|B) and P(B)
P(A) - Prob of testing positive, irrespective of having a disease or no. Therefore True Positive and False Positive.

P(A) = P(A|B) * P(B) + P(A| ~.B ) * P( ~B)
P(A) = 0.993 * 0.000025 + (1 - 0.9999) * (1 - 0.000025)
P(A) = 0.000024825 + 0.999975
P(A) = 0.0001248225

Now, we can calculate P(B|A):
P(B|A) = P(A|B) * P(B) / P(A)
P(B|A) = 0.993 * 0.000025/0.0001248225
P(B|A) = 0.1989

Therefore answer is 19.89%

In [4]:
round((0.198882413*100),2)

19.89

### ---------------------------------------------------------------------------------------------------------------------------------